tests/trainer/optimization/test_optimizers.py

```diff
@@ -405,6 +405,33 @@ def test_lr_scheduler_strict(tmpdir):
         trainer.fit(model)
 
 
+@pytest.mark.parametrize("complete_epoch", [True, False])
+@mock.patch("torch.optim.lr_scheduler.ReduceLROnPlateau.step")
+def test_lr_scheduler_strict_incomplete_epoch(step_mock, tmpdir, complete_epoch):
```

Comment:
@carmocca opinion, should I combine this test with the above? note it will be a tradeoff between readability/complexity vs. less code dupe.

Reply:
I would :)

Both together still seem readable to me.

Reply:
I combined the tests